app/Models/DatabaseDAOInterface.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+interface DatabaseDAOInterface
+{
+	/**
+	 * @return bool
+	 */
+	public function tablesAreCorrect(): bool;
+
+	/**
+	 * @param string $table
+	 * @return array
+	 */
+	public function getSchema(string $table): array;
+
+	/**
+	 * @param array $dao
+	 * @return array
+	 */
```

Comment:
Comments can be removed when there is no additional information than the native type hints